nix/modules/kld/default.nix

```
@@ -157,6 +158,10 @@ in
   config = {
     # for cli
     environment.systemPackages = [ kld-cli bitcoin-cli ];
+    environment.etc."system-info.toml".text = lib.mkDefault ''
+      git_sha = "${self.rev or "dirty"}"
+      git_commit_date = "${self.lastModifiedDate}"
+    '';
```

You can simply use an activation script instead using SSH from rust.
This is less code and also works if kld-ctl breaks during upgrade.

```suggestion
    '';
    system.activationScripts.nixos-upgrade = ''
      ${config.systemd.package}/bin/systemd-run --collect --unit nixos-upgrade echo level=info message="git_sha=${} git_commit_date=${self.lastModifiedDate}"
  '';

```

---

`CARGO_PKG_VERSION` could not be included, so the chainge is not applied.

---

`CARGO_PKG_VERSION` and bitcoind version are not included for kld-node, and DB version is not included for db-node. 

We can provide these versions in the update event, once we update nodes, we can exactly know what we updated.

---

Thank you very much, I get the idea.